subsys/net/ip/net_context.c

```diff
@@ -831,16 +831,27 @@ int net_context_bind(struct net_context *context, const struct sockaddr *addr,
 #if defined(CONFIG_NET_IPV4)
 	if (addr->sa_family == AF_INET) {
 		struct sockaddr_in *addr4 = (struct sockaddr_in *)addr;
+		struct net_if *iface = NULL;
 		struct net_if_addr *ifaddr;
-		struct net_if *iface;
 		struct in_addr *ptr;
 		int ret;
 
 		if (addrlen < sizeof(struct sockaddr_in)) {
 			return -EINVAL;
 		}
 
-		if (addr4->sin_addr.s_addr == INADDR_ANY) {
+		if (net_is_ipv4_addr_mcast(&addr4->sin_addr)) {
+			struct net_if_mcast_addr *maddr;
+
+			maddr = net_if_ipv4_maddr_lookup(&addr4->sin_addr,
+							 &iface);
+			if (!maddr) {
+				return -ENOENT;
+			}
+
+			ptr = &maddr->address.in_addr;
+
```

Comment:
remove this empty line

Reply:
Empty line would look actually better here, it actually is more crowded after removal.